debezium-core/src/test/java/io/debezium/util/CollectTest.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.util;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+import java.util.Arrays;
+import java.util.Set;
+
+/**
+ * Unit test for {@link Collect}.
+ *
+ * @author Gunnar Morling
+ */
+public class CollectTest {
+
+    public void testUnmodifiableSetForIterator() {
+        Set<Integer> values = Collect.unmodifiableSet(Arrays.asList(1, 2, 3, 42).iterator());
```

Comment:
A small note - as this is *unmodifiable* collection you should also test what will happen if you call `add` on it.